mundane-parse/src/main/scala/com/ambiata/mundane/parse/ListParser.scala

```diff
@@ -289,19 +288,21 @@ object ListParser {
   def delimitedValues[A](p: ListParser[A], delimiter: Char): ListParser[List[A]] =
     ListParser((position, state) => state match {
       case h :: t =>
-        val parsed = repeat(p).parse(Delimited.parseRow(h, delimiter).filter(_.nonEmpty))
-        parsed.map { case (_, _, as) => (position + 1, t, as) }
-
+        import scala.annotation.tailrec
+        @tailrec def traverseListParser[B](p: ListParser[B], ls: List[String], as: List[B]): Validation[String, List[B]] = ls match {
```

Comment:
This is probably generally applicable, in that the entire ListParser isn't stack safe - so it might pay to just make this a public function - might come in handy.